test/resource-limits/resource_limits.td

```diff
@@ -0,0 +1,230 @@
+ALTER SYSTEM SET max_replicas_per_cluster = 1
+
+> SHOW max_replicas_per_cluster
+1
+
+! CREATE CLUSTER REPLICA c1.r2 SIZE '1'
+contains:Replicas per cluster resource limit of 1 cannot be exceeded. Current amount is 1.
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_replicas_per_cluster = 100
+
+> SHOW max_replicas_per_cluster
+100
+
+#> CREATE CLUSTER REPLICA c1.r2 SIZE '1'
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_databases = 1
+
+> SHOW max_databases
+1
+
+! CREATE DATABASE d1
+contains:Database resource limit of 1 cannot be exceeded. Current amount is 1.
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_databases = 42
+
+> SHOW max_databases
+42
+
+> CREATE DATABASE d1
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_schemas_per_database = 2
+
+> SHOW max_schemas_per_database
+2
+
+! CREATE SCHEMA s2
+contains:Schemas per database resource limit of 2 cannot be exceeded. Current amount is 2.
+
+> CREATE SCHEMA d1.s2
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_schemas_per_database = 3
+
+> SHOW max_schemas_per_database
+3
+
+> CREATE SCHEMA s2
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_roles = 2;
+
+> SHOW max_roles
+2
+
+> CREATE ROLE joe LOGIN SUPERUSER
+
+! CREATE ROLE mike LOGIN SUPERUSER
+contains:Role resource limit of 2 cannot be exceeded. Current amount is 2.
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_roles = 3;
+
+> SHOW max_roles
+3
+
+> CREATE ROLE mike LOGIN SUPERUSER
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_secrets = 1
+
+> SHOW max_secrets
+1
+
+> CREATE SECRET secret AS 'secure_password'
+
+! CREATE SECRET password AS 'pass'
+contains:Secret resource limit of 1 cannot be exceeded. Current amount is 1.
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_secrets = 50000
+
+> SHOW max_secrets
+50000
+
+> CREATE SECRET password AS 'pass'
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_materialized_views = 1
+
+> SHOW max_materialized_views
+1
+
+> CREATE MATERIALIZED VIEW mv1 AS SELECT 1
+
+! CREATE MATERIALIZED VIEW mv2 AS SELECT 2
+contains:Materialized view resource limit of 1 cannot be exceeded. Current amount is 1.
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_materialized_views = 2
+
+> SHOW max_materialized_views
+2
+
+> CREATE MATERIALIZED VIEW mv2 AS SELECT 2
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_sources = 667
+
+> SHOW max_sources
+667
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_sinks = 44
+
+> SHOW max_sinks
+44
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM SET max_tables = 0
+
+> DROP TABLE t1
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM RESET max_tables
+
+> SHOW max_tables
+25
+
+$ postgres-execute connection=mz_system
+ALTER SYSTEM RESET ALL
+
+> SHOW max_databases
+1000
+
+> SHOW max_schemas_per_database
+1000
```

Comment:
Item No 7  You may wish to do `DROP CLUSTER` on all the clusters created in this .td file. This will make things easier if the td file is ever moved or run as part of other stuff. testdrive will issue `DROP DATABASE` at the start, which I think takes care of all the other types of database objects.

Reply:
Done.